cloudimg/ms_azure.py

```diff
@@ -285,11 +287,15 @@ def filter_object_by_tags(self, tags):
                 blobs_list = container_client.find_blobs_by_tags(
                     filter_expression=filter_expression
                 )
-                try:
-                    return blobs_list.next()
-                except StopIteration:
-                    continue
-        return
+                for tag in tags.keys():
+                    log.debug("Filtering the blob with the tag: %s" % tag)
+                    try:
+                        blobs_list = blobs_list.next()
+                    except StopIteration:
+                        log.debug("Blob with tag \"%s\" was not found" % tag)
```

Comment:
Even if you did know that, when called with N tags, find_blobs_by_tag should return N blobs with order corresponding to tags, this would still be wrong.  If, for example, you have 6 tags, but `next` raised `StopIteration` after just 5 calls, how can you possibly know it's the sixth tag which is missing and not one of the earlier ones?

Reply:
Sure.. But in this case I was assuming something wrong during this implementation, it makes no sense at all.

We're ok with the current version thus I've closed this PR